Sources/SpeziCLAID/CLAIDRuntime.swift

```diff
@@ -0,0 +1,170 @@
+        if let scriptDirectory = Bundle.main.resourcePath {
+            
+            // Add necessary paths for Python dependencies.
+            // The claid_native framework contains the implementation of the Python runtime.
+            // Normally, this is available as pip package (pip install CLAID).
+            let packageDirectory = "\(scriptDirectory)/Frameworks/claid_native_xcframework.framework/"
+            sys.path.append(packageDirectory)
+            
+            // This folder contains the PyCLAIDLoader.py, so we have to add it to the Python path.
+            let bundleDir = "\(scriptDirectory)/SpeziCLAID_SpeziCLAID.bundle/Python"
+            sys.path.append(bundleDir)
+            
+            // Add the current Bundle as path as well, enabling us to import Python files from the Resources.
+            sys.path.append(scriptDirectory)
+            
+            self.pyCLAIDLoader = Python.import("PyCLAIDLoader")
+            
+            // Ensure pyCLAID is successfully imported
+            guard let pyCLAIDLoader = self.pyCLAIDLoader else {
+                throw CLAIDError("Failed to start pyCLAID, failed to import PyCLAIDLoader, object is null.")
+            }
+            
+            // Import and add each registered Python module
+            try CLAIDRuntime.moduleRegistry.pythonModuleDetails.forEach { moduleInfo in
+                let result = pyCLAIDLoader.import_and_add_module(
+                    scriptDirectory,
+                    moduleInfo.pythonFilePath,
+                    moduleInfo.pythonModuleName
+                )
+                
+                if Bool(result) == false {
+                    throw CLAIDError("Failed to import PythonModule \(moduleInfo.pythonModuleName) from file \(moduleInfo.pythonFilePath)." +
+                                     "File not found or does not contain class '\(moduleInfo.pythonModuleName)'. Check files under \(scriptDirectory).")
+                }
+            }
+        }
+        
+    }
+    
+  
+    
+    /// Initializes the Python interpreter and starts CLAID
+    /// - Throws: CLAIDError if initialization fails
+    public func start() throws {
+        try setUpPythonDependencies()
+        try writeOutCLAIDConfig(config: self.config)
+        
+        // Start asynchronous initialization.
+        Task {
+            await self.startCLAIDFramework()
+        }
+    }
+
+        
+    /// Default initializer for the CLAIDModule
+    public init(hostName: String, userName: String, deviceId: String) {
+        // self.moduleRegistrationFunc = registerModules
+        self.hostName = hostName
+        self.userName = userName
+        self.deviceId = deviceId
+        
+        self.config = Claidservice_CLAIDConfig()
+        var hostconfig = Claidservice_HostConfig()
+        hostconfig.hostname = hostName
+        hostconfig.modules = CLAIDRuntime.moduleRegistry.moduleConfigurations
+        
+        self.config.hosts.append(hostconfig)
```

Comment:
I suspect that we might want to wait with that until the configure call so all modules are able to register their elements in the module initializers first?